src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlReflectionImporter.cs

```diff
@@ -839,19 +847,19 @@ private bool InitializeStructMembers(StructMapping mapping, StructModel model, b
             mapping.SetContentModel(textAccesor, hasElements);
             if (isSequence)
             {
-                Hashtable ids = new Hashtable();
+                var ids = new HashSet<int>();
                 for (int i = 0; i < members.Count; i++)
                 {
                     MemberMapping member = (MemberMapping)members[i];
                     if (!member.IsParticle)
                         continue;
                     if (member.IsSequence)
                     {
-                        if (ids[member.SequenceId] != null)
+                        if (ids.Contains(member.SequenceId))
```

Comment:
Now that `ids` is `HashSet<int>`, the `Contains` and subsequent `Add` calls can be replaced with a single call to `Add` inside the `if` check, since `HashSet<T>.Add` returns `false` if the item was already in the set. Avoids multiple lookups.

```csharp
if (!ids.Add(member.SequenceId))
{
    throw new InvalidOperationException(SR.Format(SR.XmlSequenceUnique, member.SequenceId.ToString(CultureInfo.InvariantCulture), "Order", member.Name));
}
```